retroshare-gui/src/gui/TheWire/WireDialog.cpp

```diff
@@ -91,6 +90,54 @@ WireDialog::WireDialog(QWidget *parent)
 
 	// load settings
 	processSettings(true);
+
+    // Needs to be asynced because this function is called by another thread!
+    rsEvents->registerEventsHandler(
+                [this](std::shared_ptr<const RsEvent> event)
+    { RsQThreadUtils::postToObject([=]() { handleEvent_main_thread(event); }, this ); },
+                mEventHandlerId, RsEventType::WIRE );
+}
+
+void WireDialog::handleEvent_main_thread(std::shared_ptr<const RsEvent> event)
+{
+    const RsWireEvent *e = dynamic_cast<const RsWireEvent*>(event.get());
+
+    if(e)
+    {
+
+#ifdef GXSWIRE_DEBUG
+                RsDbg() << " Refreshing the feed if there is a matching event. "<< std::endl;
+#endif
+
+        // The following switch statements refresh the wire feed whenever there is a new event
+        switch(e->mWireEventCode)
+        {
+        case RsWireEventCode::NEW_POST:
+            refreshGroups();
+            break;
```

Comment:
For code readability and simplification, plz remove the break and calls to refreshGorup(), except the last one.